src/main/java/uk/gov/hmcts/reform/em/stitching/service/CdamService.java

```diff
@@ -0,0 +1,120 @@
+            .parallel()
+            .map(unchecked(bundleDocument -> downloadFile(documentTask.getJwt(), documentTask.getServiceAuth(), bundleDocument)));
+    }
+
+    public Pair<BundleDocument, FileAndMediaType> downloadFile(String auth, String serviceAuth, BundleDocument bundleDocument) throws
+            IOException, DocumentTaskProcessingException {
+        String docId = bundleDocument.getDocumentURI().substring(bundleDocument.getDocumentURI().lastIndexOf('/') + 1);
+        UUID documentId = UUID.fromString(docId);
+        ResponseEntity<Resource> response =  caseDocumentClientApi.getDocumentBinary(auth, serviceAuth, documentId);
+        HttpStatus status = null;
+
+        if (Objects.nonNull(response)) {
+            status = response.getStatusCode();
+            var byteArrayResource = (ByteArrayResource) response.getBody();
+            if (Objects.nonNull(byteArrayResource)) {
+                try (var inputStream = byteArrayResource.getInputStream()) {
+                    Document document = caseDocumentClientApi.getMetadataForDocument(auth, serviceAuth, documentId);
+                    var originalDocumentName = document.originalDocumentName;
+                    var fileType = FilenameUtils.getExtension(originalDocumentName);
+                    var fileName = "document." + fileType;
+                    File file = copyResponseToFile(inputStream, fileName);
+                    return Pair.of(bundleDocument,
+                        new FileAndMediaType(file, okhttp3.MediaType.get(document.mimeType)));
+                }
+            }
+        }
+
+        throw new DocumentTaskProcessingException(String.format("Could not access the binary. HTTP response: %s",
+                status));
+    }
+
+    private File copyResponseToFile(InputStream inputStream, String fileName) throws DocumentTaskProcessingException {
+        try {
+
+            var tempDir = Files.createTempDirectory("pg",
+                PosixFilePermissions.asFileAttribute(EnumSet.allOf(PosixFilePermission.class)));
+            var tempFile = new File(tempDir.toAbsolutePath().toFile(), fileName);
+
+            Files.copy(inputStream, tempFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+            return tempFile;
+        } catch (IOException e) {
+            throw new DocumentTaskProcessingException("Could not copy the file to a temp location", e);
+        }
+    }
+
+    public void uploadDocuments(File file, DocumentTask documentTask) throws DocumentTaskProcessingException {
+
+        try {
+            ByteArrayMultipartFile multipartFile =
+                ByteArrayMultipartFile.builder()
+                    .content(FileUtils.readFileToByteArray(file))
+                    .name(file.getName() + ".pdf")
+                    .contentType(MediaType.valueOf("application/pdf"))
+                .build();
+
+            DocumentUploadRequest documentUploadRequest = new DocumentUploadRequest(Classification.PUBLIC.toString(),
```

Comment:
can we put some inline doco as to why this is default of PUBLIC? maybe a link to wiki etc
possibly this should be a constant